jsign-crypto/src/main/java/net/jsign/jca/VenafiSigningService.java

```diff
@@ -0,0 +1,269 @@
+                return new byte[]{0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60, (byte)0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01, 0x05, 0x00, 0x04, 0x20}; // SHA256
+            case 65: case 4165:
+                return new byte[]{0x30, 0x41, 0x30, 0x0d, 0x06, 0x09, 0x60, (byte)0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x02, 0x05, 0x00, 0x04, 0x30}; // SHA384
+            case 66: case 4166:
+                return new byte[]{0x30, 0x51, 0x30, 0x0d, 0x06, 0x09, 0x60, (byte)0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x03, 0x05, 0x00, 0x04, 0x40}; // SHA512
+            default: 
+                return new byte[]{0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60, (byte)0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01, 0x05, 0x00, 0x04, 0x20}; // SHA256
+        }
+    }
+
+
+    /**
+     * Creates a new Venafi CodeSign Protect service.
+     *
+     * @param endpoint         the Venafi API endpoint (for example <tt>https://demo.venafitpp.local/vedhsm/api/sign/</tt>)
+     * @param credentials      the Venafi credentials
+     */
+    public VenafiSigningService(String endpoint, VenafiCredentials credentials) throws IOException {
+        if (!endpoint.startsWith("http")) {
+            endpoint = "https://" + endpoint;
+        }
+        this.endpoint = endpoint;
+        this.credentials = credentials;
+        String token = credentials.getSessionToken(endpoint);
+        this.client = new RESTClient(endpoint)
+            .authentication(conn -> conn.setRequestProperty("Authorization", "Bearer " + token ))
+            .errorHandler(response -> response.get("error") + ": " + response.get("error_description"));
+                
+    }
+
+    @Override
+    public String getName() {
+        return "Venafi";
+    }
+
+    private void loadKeyStore(String alias) throws KeyStoreException {
+        if (certificates.isEmpty()) {
+            try {
+                Map<String, Object> request = new LinkedHashMap<>();
+                request.put("EnvironmentFilter", new Integer[] { 0});
+                request.put("ObjectTypeFilter", new Integer[] { 1});
+                request.put("IncludeChains", true);
+                request.put("LabelFilter", new String[] { alias });
+        
+                Map<String, ?> response = client.post("/vedhsm/api/getobjects", JsonWriter.format(request));
+                
+                Object[] keys = (Object[]) response.get("Certificates");
+                for (Object key : keys) {
+                    String name = (String) ((Map) key).get("Label");
+                    KeyId = (String) ((Map) key).get("KeyId");
+                    certificates.put(name, (Map<String, ?>) key);
+                }
+            } catch (IOException e) {
+                throw new KeyStoreException("Unable to retrieve the Venafi keystore with alias: " + alias, e);
+            }
+        }
+    }
+
+    @Override
+    public List<String> aliases() throws KeyStoreException {
+        return new ArrayList<>(certificates.keySet());
```

Comment:
Is there an API call to initialize that could be used to initialize the certificates available? Otherwise this method will return an empty list if `getCertificateChain()` hasn't been called yet.